### PR TITLE
Fixes #2255: leaked transactions in ConsistentKeyLocker

### LINKED ISSUE #2255: Leaked transactions in ConsistentKeyLocker
Transactions created in ConsistentKeyLocker.overrideTimestamp() are neither committed or rolled back by the callers. This results in leaked transactions. ConsistentKeyLocker needs to be updated to commit or rollback the created transactions.
 
Relevant code from ConsistentKeyLocker.java given below.

https://github.com/JanusGraph/janusgraph/blob/9ba0b2d850066a32551f9da4af0780f643288376/janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java#L549-L554

https://github.com/JanusGraph/janusgraph/blob/9ba0b2d850066a32551f9da4af0780f643288376/janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java#L384-L388

https://github.com/JanusGraph/janusgraph/blob/9ba0b2d850066a32551f9da4af0780f643288376/janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java#L400-L403

https://github.com/JanusGraph/janusgraph/blob/9ba0b2d850066a32551f9da4af0780f643288376/janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java#L535-L539

### PR DESCRIPTION
**Pull request by mneethiraj**
Fixed transaction leaks in ConsistentKeyLocker. Updated unit tests to detect leaked transaction.

### CODE REVIEW

**Comment by mneethiraj** [comment]
CI failure seems to be unrelated to this commit. Can I please get help with this?

Thanks!

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project janusgraph-driver: Compilation failure: Compilation failure: 
[ERROR] error reading /home/travis/.m2/repository/org/testcontainers/testcontainers/1.14.3/testcontainers-1.14.3.jar; error in opening zip file
[ERROR] error reading /home/travis/.m2/repository/org/testcontainers/testcontainers/1.14.3/testcontainers-1.14.3.jar; cannot read zip file
[ERROR] /home/travis/build/JanusGraph/janusgraph/janusgraph-driver/src/test/java/org/janusgraph/core/JanusGraphContainer.java:[25,37] package org.testcontainers.containers does not exist
[ERROR] /home/travis/build/JanusGraph/janusgraph/janusgraph-driver/src/test/java/org/janusgraph/core/JanusGraphContainer.java:[26,51] package org.testcontainers.containers.wait.strategy does not exist
[ERROR] /home/travis/build/JanusGraph/janusgraph/janusgraph-driver/src/test/java/org/janusgraph/core/JanusGraphContainer.java:[28,42] cannot find symbol
[ERROR]   symbol: class GenericContainer
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :janusgraph-driver

```

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java [hunk 1]
```diff
@@ -381,13 +381,28 @@ private WriteResult tryWriteLockOnce(StaticBuffer key, StaticBuffer del, StoreTr
...
+
+            newTx.commit();
+            newTx = null;
         } catch (BackendException e) {
             log.debug("Lock write attempt failed with exception", e);
             t = e;
+        } finally {
+            if (newTx != null) {
+                try {
+                    log.error("Transaction is still open! Rolling back", new Throwable());
```
Comment (li-boxuan) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java:399]: Can you remove this logging statement? We usually don't log when we do a rollback in the final section.

**Review by li-boxuan — COMMENTED** [review]
Thanks for your contribution! Looks good to me in general, I just left a short comment.

Can you also retarget this PR at the v0.5 branch? v0.5 branch is the oldest version that we support. After this PR is merged, we will merge v0.5 to master.

**Comment by li-boxuan** [comment]
Closing this PR in favor of https://github.com/JanusGraph/janusgraph/pull/2258

**Pull request closed without merging**